db/db_iter.cc

```diff
@@ -388,8 +400,17 @@ bool DBIter::FindNextUserEntryInternal(bool skipping_saved_key,
         // We're looking for the next user-key but all we see are the same
         // user-key with decreasing sequence numbers. Fast forward to
         // sequence number 0 and type deletion (the smallest type).
-        AppendInternalKey(&last_key, ParsedInternalKey(saved_key_.GetUserKey(),
-                                                       0, kTypeDeletion));
+        if (timestamp_size_ == 0) {
+          AppendInternalKey(
+              &last_key,
+              ParsedInternalKey(saved_key_.GetUserKey(), 0, kTypeDeletion));
+        } else {
+          std::string min_ts(timestamp_size_, static_cast<char>(0));
```

Comment:
We assume that all 0 string is the smallest timestamp. Is it fair to make the assumption? Or should we return a min timestamp in comparator?

Reply:
I think in usual cases it is ok, but I can add a new virtual function to `Comparator`, e.g. `Comparator::GetMinTimestamp() const`. I have only one concern. I imagine we are going to add a `clock` object as db options in the future. Should `GetMinTimestamp()` be part of `clock` or `comparator`.

Reply:
Then maybe add it to Clock eventually. We can leave it as 0 now and have a comment here.

Reply:
SGTM.